addons/payment_alipay/models/payment_acquirer.py

```diff
@@ -0,0 +1,78 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import logging
+
+from hashlib import md5
+from werkzeug import urls
+
+from odoo import api, fields, models, _
+from odoo.tools.float_utils import float_compare
+from odoo.addons.payment_alipay.controllers.main import AlipayController
+from odoo.addons.payment.models.payment_acquirer import ValidationError
+
+_logger = logging.getLogger(__name__)
+
+
+class PaymentAcquirer(models.Model):
+    _inherit = 'payment.acquirer'
+
+    provider = fields.Selection(selection_add=[
+        ('alipay', 'Alipay')
+    ], ondelete={'alipay': 'set default'})
+    alipay_payment_method = fields.Selection([
+        ('express_checkout', 'Express Checkout (only for Chinese Merchant)'),
+        ('standard_checkout', 'Cross-border'),
+    ], string='Account', default='express_checkout',
+        help="  * Cross-border: For the Overseas seller \n  * Express Checkout: For the Chinese Seller")
+    alipay_merchant_partner_id = fields.Char(
+        string='Merchant Partner ID', required_if_provider='alipay', groups='base.group_user',
+        help='The Merchant Partner ID is used to ensure communications coming from Alipay are valid and secured.')
+    alipay_md5_signature_key = fields.Char(
+        string='MD5 Signature Key', required_if_provider='alipay', groups='base.group_user',
+        help="The MD5 private key is the 32-byte string which is composed of English letters and numbers.")
+    alipay_seller_email = fields.Char(string='Alipay Seller Email', groups='base.group_user')
+
+    @api.model
+    def _get_alipay_urls(self):
+        """ Alipay URLS """
+        environment = 'prod' if self.state == 'enabled' else 'test'
+        if environment == 'prod':
+            return 'https://mapi.alipay.com/gateway.do'
+        return 'https://openapi.alipaydev.com/gateway.do'
```

Comment:
Nitpicking: environment is never evaluated as 'test'.
```suggestion
        if self.state == 'enabled':
            return 'https://mapi.alipay.com/gateway.do'
        else:  # test
            return 'https://openapi.alipaydev.com/gateway.do'
```